pkg/cluster/backup_manager.go

```diff
@@ -240,3 +240,27 @@ func (bm *backupManager) upgradeIfNeeded() error {
 	}
 	return k8sutil.PatchDeployment(bm.config.KubeCli, ns, n, uf)
 }
+
+func (bm *backupManager) delete() error {
```

Comment:
It's confusing to have both delete() and cleanup().
Better naming this? Maybe deleteBackupSidecar()?